### PR TITLE
Ensure the Nebula device exists before attempting to bind to the Nebula IP

### PR DESCRIPTION
**Pull request by brad-defined**
As reported in https://github.com/slackhq/nebula/issues/243 and https://github.com/slackhq/nebula/issues/314, internal services that are configured to bind only to the Nebula IP sometimes fail because they attempt to bind to the Nebula interface before it's created. This diff moves the start of the SSH, DNS, and Stats internal services until after we're certain that the Nebula interface is created.

### CODE REVIEW

**Comment by brad-defined** [comment]
I rebased against master, and re-tested. I discovered that I never actually fixed the race in the stats case, so that code moved around a little bit. (I didn't wrap the Listen call in the delayed function, so it still failed when bound only to the Nebula IP. Oops.)

I also updated the Control struct to have named variables for the delayed startup functions, instead of the more generic array.

File: sshd/server.go [hunk 1]
```diff
@@ -93,6 +93,7 @@ func (s *SSHServer) Run(addr string) error {
 	var err error
 	s.listener, err = net.Listen("tcp", addr)
 	if err != nil {
+		s.l.WithField("err", err).Warn("Error starting SSH server listener, shutting down")
```
Comment (nbrownus) [inline sshd/server.go:96]: Caller should be the one to log

File: stats.go [hunk 2]
```diff
@@ -17,24 +17,35 @@ import (
...
 
 	interval := c.GetDuration("stats.interval", 0)
 	if interval == 0 {
-		return fmt.Errorf("stats.interval was an invalid duration: %s", c.GetString("stats.interval", ""))
+		return nil, fmt.Errorf("stats.interval was an invalid duration: %s", c.GetString("stats.interval", ""))
 	}
 
+	var startFn func()
 	switch mType {
 	case "graphite":
```
Comment (nbrownus) [inline stats.go:36]: I recall an old debate regarding this, I think, will need to noodle

File: sshd/server.go (hunk 1 above)
Comment (brad-defined) [reply sshd/server.go]: Moved the log line, and verified that I saw the error log when misconfiguring the ssh listener.

**Review by nbrownus — APPROVED** [review]
Looks good, also closes #393

**Pull request merged**